[#25] 약 전송 요청 확인뷰 UI 구현

EunHee-Jeong opened the pull request:
## 🌴 PR 요약
알림 탭 두번째 뷰 (전송받은 약 상세 정보) 화면을 구성했습니다 ~!
<!-- PR의 내용을 요약해주세요. -->

🌱 작업한 브랜치

- feature/#25

🌱 작업한 내용

- 컨테이너 뷰 UI 구성하기 (상단바, 헤더, 버튼 - 고정)
- 컬렉션 뷰 UI 구성하기 (약 상세 보기와 정보 - 스크롤)
- 더 자세한 내용은 커밋에 설명 달아놓았습니다...

## 📸 스크린샷

| 기능 |   스크린샷   |
| :--: | :----------: |
| GIF  | <img src="https://user-images.githubusercontent.com/70689381/149492598-67a90bea-fb91-4ceb-a2ae-9b068e182dbe.gif" width="300"> |

## 📮 관련 이슈

- Resolved: #25

Commit:
[Add] 컨테이너 뷰 파일 생성 (#25)

Commit:
[Del] NavigationBarView 파일 제거 (#25)

- xib 파일 충돌 문제

Commit:
[Style] 상단바 구현 (#25)

- 방식 변경: 코드로 구현

Commit:
[Style] 컨테이너 뷰 오토레이아웃 (#25)

Commit:
[Style] 바텀뷰 구현 (#25)

- 버튼 스택뷰에 묶어서 넣어주고 오토레이아웃 잡음

Commit:
[Style] 탑뷰 구현 (#25)

- 보낸 사람 정보 표시

Commit:
[Add] 약 색깔 아이콘 이미지 에셋 추가 (#25)

Commit:
[Edit] 이미지 에셋 추가와 네이밍 수정 (#25)

Commit:
[Style] 전송받은 약 정보 셀 UI 구성 (#25)

- 상세정보 ➡️ 스택뷰 이용, 시간 추가에 제한 없기 때문에 bottom constraint 잡지 않음

Commit:
[Feat] 전송받은 약 정보 UI 구현 (#25)

- 구조: UIView + UICollectionView + UIView

Commit:
[Edit] 상단바 UI 수정 (#25)

Commit:
[Add] FooterCollectionViewCell 파일 생성 (#25)

- UICollectionViewCell 이용

Commit:
[Style] 푸터뷰 UI 구성 (#25)

Commit:
[Feat] 푸터뷰 등록과 레이아웃 수정 (#25)

- 전체 레이아웃 일정 부분 조정
- 약 정보, 푸터뷰 간격을 각각 처리해줌. (코드 디폴트 간격: 11, 푸터뷰: UIView 안에 넣고 오토레이아웃으로 +5)

Taehyeon-Kim commented:
1. 디자인 상 `효영님이 전송한 약이에요` 라벨 부분도 스크롤할때 같이 올라가야 하네요 확인부탁드립니다~
2. 시간 부분 2줄 이상 일 때도 정상적으로 UI 확인되는지 스크린 샷 부탁드려요!
3. 시간 한 줄 일 때, `1주에 한 번` 요 부분과의 간격, 그리고 왼쪽에 선 길이 디자인 시안과 일치하는지 확인부탁드립니다!
4. 🚨 Conflict 난 부분도 확인 필요! @EunHee-Jeong

Taehyeon-Kim commented on SobokSobok/SobokSobok/Data/Model/SendInfoListDataModel.swift:
```diff
@@ -0,0 +1,31 @@
+import UIKit
+
+struct SendInfoListData {
+    
```
줄 제거하기

Taehyeon-Kim commented on SobokSobok/SobokSobok/Presentation/Notice/FooterCollectionViewCell.swift:
```diff
@@ -0,0 +1,17 @@
+
+import UIKit
+
+class FooterCollectionViewCell: UICollectionViewCell {
```
```suggestion
final class FooterCollectionViewCell: UICollectionViewCell {
```

Taehyeon-Kim commented on SobokSobok/SobokSobok/Presentation/Notice/SendInfoViewController.swift:
```diff
@@ -0,0 +1,118 @@
+    }
+    let navigationTitleLabel = UILabel().then {
+        $0.text = "전송 받은 약"
+        $0.font = .systemFont(ofSize: 17)
```
폰트 추가해놓았으니 일괄적용 부탁드립니다~

```swift
  titleLabel.setTypoStyle(
      font: TypoStyle.body5.font,
      kernValue: TypoStyle.body5.labelDescription.kern,
      lineSpacing: TypoStyle.body5.labelDescription.lineHeight
  )
```

figma에 보면 텍스트 클릭
<img width="221" alt="스크린샷 2022-01-14 오후 6 45 33" src="https://user-images.githubusercontent.com/61109660/149494726-a77cf0f4-69a7-4c05-bca6-4b74e0ddcb29.png">
했을 때 다음과 같이 정보가 나오는데

body 정보만 바꿔서 적용하면 됩니다.
주의해야할 점은 Label에 직접 적용한다는 부분!

EunHee-Jeong replied:
body 정보 바꿔서 적용하기 / Label에 직접 적용... 넵!! (very kind 👍)

Taehyeon-Kim reviewed (commented):
GOOD ~ 우선 피드백 드린 부분 확인 부탁드립니다!

EunHee-Jeong commented:
> 

1. 고정인줄 알았네여ㅜ 수정하겠습니당 !
2.
 <img src="https://user-images.githubusercontent.com/70689381/149494882-bfc5dd2a-fed7-4a52-853e-2218ae783d40.png" width="300">
3. 오 이거 질문인데 같이 PR 올리는 거 까먹었네요... 오토레이아웃으로 잡아줘도 적용이 안 되더라구요...
<img src="https://user-images.githubusercontent.com/70689381/149496399-5694f303-4c4c-4950-803d-bce589cc2e5d.png" width="300">

Taehyeon-Kim commented:
> 3. 오 이거 질문인데 같이 PR 올리는 거 까먹었네요... 오토레이아웃으로 잡아줘도 적용이 안 되더라구요...
> <img alt="" width="300" src="https://user-images.githubusercontent.com/70689381/149496399-5694f303-4c4c-4950-803d-bce589cc2e5d.png">

셀마다 높이가 달라지는 상황이 생길텐데 이 부분은 조금 챌린징한 요소가 될 것 같습니다.  
우선 다음 블로그에서 내용을 참고하시면 좋을 것 같아요!
https://corykim0829.github.io/ios/UICollectionViewCell-dynamic-height/#

Commit:
[Edit] 코드리뷰 반영과 구문 수정 (#25)

- if-else문 ➡️ switch-case문으로 변경
- 헤더 ➡️ SenderInfoCVC 파일 만들어서 스크롤 영역에 포함되게 해줌

Commit:
[Fix] conflicts 해결 (#25)

seondal commented on SobokSobok/SobokSobok/Presentation/Notice/SendInfoViewController.swift:
```diff
@@ -0,0 +1,124 @@
+        }
+    }
+    
+    private func setConstraints() {
+        navigationView.snp.makeConstraints {
+            $0.top.leading.trailing.equalToSuperview()
+            $0.height.equalTo(102)
+        }
+        
+        navigationTitleLabel.snp.makeConstraints {
+            $0.centerX.equalTo(navigationView)
+            $0.bottom.equalTo(navigationView).inset(20)
+        }
+        
+        xButton.snp.makeConstraints {
+            $0.leading.equalTo(navigationView).inset(1)
+            $0.centerY.equalTo(navigationTitleLabel)
+        }
+    }
```
코드베이스로 진행하신건가용??

EunHee-Jeong replied:
넴 상단바 부분입니당 !

seondal reviewed (dismissed):
👍

Taehyeon-Kim commented:
> > 3. 오 이거 질문인데 같이 PR 올리는 거 까먹었네요... 오토레이아웃으로 잡아줘도 적용이 안 되더라구요...
> > 
> > <img alt="" width="300" src="https://user-images.githubusercontent.com/70689381/149496399-5694f303-4c4c-4950-803d-bce589cc2e5d.png">
> 
> 셀마다 높이가 달라지는 상황이 생길텐데 이 부분은 조금 챌린징한 요소가 될 것 같습니다. 우선 다음 블로그에서 내용을 참고하시면 좋을 것 같아요! https://corykim0829.github.io/ios/UICollectionViewCell-dynamic-height/#

이부분 메인에서도 진행중인데 해결되면 공유드릴게요
@EunHee-Jeong @seondal @seungchan2

Commit:
[Style] 헤더뷰 CGSize 추가 (#25)

seungchan2 commented on SobokSobok/SobokSobok/Presentation/Notice/SendInfoViewController.swift:
```diff
@@ -0,0 +1,126 @@
+            view.addSubview($0)
+        }
+        [refuseButton, acceptButton].forEach {
+            $0.cornerRadius = 12
```
저번에 makeCorner~ 만든 거 쓰는 것도 좋을 듯!!

Commit:
[Edit] 셀 빈공간 문제 해결 (#25)

- 코드로 잡아준 부분 때문에 스크롤 뷰 밀리는 문제 ➡️ Top Space : SuperView(0) 으로 맞춰줌

Commit:
[Fix] Git Conflicts 해결 (#25)

# 본문은 위에 작성
######## 본문은 한 줄에 최대 72 글자까지만 입력 ########################### -> |
# 꼬릿말은 아래에 작성: ex) #이슈 번호
# --- COMMIT END ---
#- ✅[CHORE] : 코드 수정, 내부 파일 수정
#- ✨[FEAT] : 새로운 기능 구현
#- ➕[ADD] : Feat 이외의 부수적인 코드 추가, 라이브러리 추가, 새로운 파일 생성 시
#- 🍱[ADD] : 에셋 추가
#- 🚑️[HOTFIX] : issue나, QA에서 급한 버그 수정에 사용
#- 🔨[FIX] : 버그, 오류 해결
#- ⚰️ [DEL] : 쓸모없는 코드 삭제
#- 📝[DOCS] : README나 WIKI 등의 문서 개정
#- 💄[MOD] : storyboard 파일,UI 수정한 경우
#- ✏️ [CORRECT] : 주로 문법의 오류나 타입의 변경, 이름 변경 등에 사용합니다.
#- 🚚[MOVE] : 프로젝트 내 파일이나 코드의 이동
#- ⏪️[RENAME] : 파일 이름 변경이 있을 때 사용합니다.
#- ⚡️[IMPROVE] : 향상이 있을 때 사용합니다.
#- ♻️[[REFACTOR] : 전면 수정이 있을 때 사용합니다
#- 🔀[MERGE]: 다른브렌치를 merge 할 때 사용합니다.
# ------------------
#     제목 첫 글자를 대문자로
#     제목은 명령문으로
#     제목 끝에 마침표(.) 금지
#     제목과 본문을 한 줄 띄워 분리하기
#     본문은 "어떻게" 보다 "무엇을", "왜"를 설명한다.
#     본문에 여러줄의 메시지를 작성할 땐 "-"로 구분
# ------------------

seungchan2 commented on SobokSobok/SobokSobok/Presentation/Notice/SendInfoCollectionViewCell.swift:
```diff
@@ -0,0 +1,34 @@
+    @IBOutlet weak var timeLabel: UILabel!
+    @IBOutlet weak var editButton: UIButton!
+    
+    // MARK: - View Life Cycle
```
안쓰면 지워주셔도 될 듯 .//?

Commit:
[Edit] 코드리뷰 반영과 UI 부분 수정 (#25)

- UI 부분 수정: Edit 버튼색 설정(gray800), 헤더뷰오토레이아웃 조정 (top 48, 상단바 타이틀에 맞춤)

Taehyeon-Kim committed:
Merge branch 'develop' into feature/#25

seungchan2 reviewed (approved):
고생 ~
